models/url_pydantic_models.py

```diff
@@ -1,12 +1,16 @@
 from pydantic import BaseModel
+from datetime import datetime
+from typing import Optional
 # Define your models here
 
 
 class URLRequest(BaseModel):
     url: str
+    custom_url: Optional[str] = None
+    length: int = 6 #do i need to enforce a minimum? max?
```

Comment:
you can use Pydantic’s constr type to enforce length constraints. 

For example: `custom_url: Optional[constr(min_length=3, max_length=10)] = None`

This sets a minimum of 3 and a maximum of 10 characters to ensure the custom URLs remain short and user-friendly.